ui/app/pages/connected-sites/connected-sites.component.js

```diff
@@ -42,73 +45,129 @@ export default class ConnectSites extends Component {
     })
   }
 
-  clearSitePendingDisconnect = () => {
+  clearPendingDisconnect = () => {
     this.setState({
       sitePendingDisconnect: null,
     })
   }
 
-  disconnect = () => {
+  disconnectAccount = () => {
     const { disconnectAccount } = this.props
     const { sitePendingDisconnect } = this.state
 
     disconnectAccount(sitePendingDisconnect.domainKey)
-    this.clearSitePendingDisconnect()
+    this.clearPendingDisconnect()
+  }
+
+  disconnectAllAccounts = () => {
+    const { disconnectAllAccounts } = this.props
+    const { sitePendingDisconnect } = this.state
+
+    disconnectAllAccounts(sitePendingDisconnect.domainKey)
+    this.clearPendingDisconnect()
   }
 
-  renderConnectedSites () {
+  renderConnectedSitesList () {
     return (
       <ConnectedSitesList
         connectedDomains={this.props.connectedDomains}
-        onDisconnectSite={this.setSitePendingDisconnect}
+        onDisconnect={this.setPendingDisconnect}
       />
     )
   }
 
-  render () {
-    const { accountLabel, history, legacyExposeAccount, tabToConnect, connectedDomains } = this.props
+  renderConnectedSitesPopover () {
+
+    const {
+      accountLabel,
+      closePopover,
+      connectedDomains,
+      legacyExposeAccount,
+      tabToConnect,
+    } = this.props
     const { t } = this.context
+
+    return (
+      <Popover
+        title={t('connectedSites')}
+        subtitle={connectedDomains.length
+          ? t('connectedSitesDescription', [accountLabel])
+          : t('connectedSitesEmptyDescription', [accountLabel])
+        }
+        onClose={closePopover}
+        footer={
+          tabToConnect
+            ? (
+              <div
```

Comment:
Hm, I see. I thought I saw a comment where someone recommended divs over anchors, which is why I did it. Your argument makes sense though; I'll edit it.

Reply:
I don't recall what complications using an anchor brings. But it is an interactive element, so that's one better than a `div` at least.

You could try a button instead if you want? That'd be fine too. I assume an anchor was chosen because it was easier to style the way we intended here, but we could use a button to and overwrite the default styles.

Reply:
Converted to anchors in b397aa5